app/controllers/miq_policy_controller/alerts.rb

```diff
@@ -52,10 +52,10 @@ def alert_delete
     alerts = []
     # showing 1 alert, delete it
 
-    if params[:id].nil? || MiqAlert.find_by_id(params[:id]).nil?
+    if params[:id].nil? || MiqAlert.find_by(:id => params[:id]).nil?
       add_flash(_("%{models} no longer exists") % {:models => ui_lookup(:model => "MiqAlert")},
                 :error)
-    elsif MiqAlert.find_by_id(params[:id]).read_only
+    elsif MiqAlert.find_by(:id => params[:id]).read_only
```

Comment:
this one should be converted to `find(params[:id])` given the `.read_only` call assumes non-null